### ISSUE TITLE
[JENKINS-36638] SSE events not coming through to the client

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#3006]
<p>Looks like SSE is acting a bit dodgy again.</p>

<p><b>Steps to reproduce</b><br/>
1. Open up the Activity tab of a pipeline in one tab<br/>
2. Open a new browser tab and run the same pipeline<br/>
3. You don't see the new run appear in the first tab</p>

<p>I watched in classic to ensure that it was in the running state so it is not <a href="https://issues.jenkins.io/browse/JENKINS-36209" title="Queued items should show on the activity tab" class="issue-link" data-issue-key="JENKINS-36209"><del>JENKINS-36209</del></a></p>

<p>Not sure if related - there is an error in the console:</p>
<blockquote>
<p>GET <a href="https://ci.blueocean.io/sse-gateway/listen/jenkins_blueocean" class="external-link" target="_blank" rel="nofollow noopener">https://ci.blueocean.io/sse-gateway/listen/jenkins_blueocean</a> net::ERR_INCOMPLETE_CHUNKED_ENCODING</p></blockquote>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638" target="_blank">SSE events not coming through to the client</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tfennelly.png" title="tfennelly's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly">tfennelly</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: blueocean-plugin
<li><b>resolution</b>: Won't Fix
<li><b>resolved</b>: 2016-07-27T07:24:31+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 3
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Looks like SSE is acting a bit dodgy again.</p>

<p><b>Steps to reproduce</b>
1. Open up the Activity tab of a pipeline in one tab
2. Open a new browser tab and run the same pipeline
3. You don't see the new run appear in the first tab</p>

<p>I watched in classic to ensure that it was in the running state so it is not <a href="https://issues.jenkins.io/browse/JENKINS-36209" title="Queued items should show on the activity tab" class="issue-link" data-issue-key="JENKINS-36209"><del>JENKINS-36209</del></a></p>

<p>Not sure if related - there is an error in the console:</p>
<blockquote>
<p>GET <a href="https://ci.blueocean.io/sse-gateway/listen/jenkins_blueocean" class="external-link" target="_blank" rel="nofollow noopener">https://ci.blueocean.io/sse-gateway/listen/jenkins_blueocean</a> net::ERR_INCOMPLETE_CHUNKED_ENCODING</p></blockquote></pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tfennelly.png" title="tfennelly's avatar" /> tfennelly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=264046&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264046">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Fairly sure the ERR_INCOMPLETE_CHUNKED_ENCODING error in the console has nothing to do with the error described in the description. The ERR_INCOMPLETE_CHUNKED_ENCODING will go away once we do <a href="https://issues.jenkins.io/browse/JENKINS-35175" title="Make SSE connection keep-alive configurable" class="issue-link" data-issue-key="JENKINS-35175"><del>JENKINS-35175</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>Fairly sure the ERR_INCOMPLETE_CHUNKED_ENCODING error in the console has nothing to do with the error described in the description. The ERR_INCOMPLETE_CHUNKED_ENCODING will go away once we do <a href="https://issues.jenkins.io/browse/JENKINS-35175" title="Make SSE connection keep-alive configurable" class="issue-link" data-issue-key="JENKINS-35175"><del>JENKINS-35175</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tfennelly.png" title="tfennelly's avatar" /> tfennelly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=264047&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264047">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Would be good to know if this error happens locally, or just on ci.blueocean.io. I wonder if it's related to the scenario described in <a href="https://issues.jenkins.io/browse/JENKINS-36238" title="Server-side store and forward SSE events during a reconnect" class="issue-link" data-issue-key="JENKINS-36238"><del>JENKINS-36238</del></a>.</p>
</pre>
</details>

</ul>
</details>

<p>Would be good to know if this error happens locally, or just on ci.blueocean.io. I wonder if it's related to the scenario described in <a href="https://issues.jenkins.io/browse/JENKINS-36238" title="Server-side store and forward SSE events during a reconnect" class="issue-link" data-issue-key="JENKINS-36238"><del>JENKINS-36238</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=264053&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264053">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I saw this on blueocean.io but did not test it locally </p>
</pre>
</details>

</ul>
</details>

<p>I saw this on blueocean.io but did not test it locally </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=264090&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264090">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This may not be the right ticket for this comment but I want to be sure to capture it. Are we sure that the SSE events aren't actually coming through, or is it that the UI just isn't updating itself?</p>

<p>While debugging <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a> I noticed some issues with UI updates. SEE events were coming through ok after a run, but the UI was not updating. The SSE event we receive for a nested job looks like this:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">{
    <span class="code-quote">"jenkins_object_type"</span>: <span class="code-quote">"org.jenkinsci.plugins.workflow.job.WorkflowJob"</span>,
    <span class="code-quote">"job_run_status"</span>: <span class="code-quote">"QUEUED"</span>,
    <span class="code-quote">"job_name"</span>: <span class="code-quote">"myfolder/blueocean-demo-project/feature%2Fnewhotness"</span>,
    <span class="code-quote">"jenkins_org"</span>: <span class="code-quote">"jenkins"</span>,
    <span class="code-quote">"job_run_queueId"</span>: <span class="code-quote">"23"</span>,
    <span class="code-quote">"job_ismultibranch"</span>: <span class="code-quote">"<span class="code-keyword">true</span>"</span>,
    <span class="code-quote">"jenkins_object_name"</span>: <span class="code-quote">"myfolder/blueocean-demo-project/feature%2Fnewhotness"</span>,
    <span class="code-quote">"blueocean_job_rest_url"</span>: <span class="code-quote">"/<span class="code-keyword">rest</span>/organizations/jenkins/pipelines/blueocean-demo-project/branches/feature%2Fnewhotness/"</span>,
    <span class="code-quote">"blueocean_job_branch_name"</span>: <span class="code-quote">"feature%2Fnewhotness"</span>,
    <span class="code-quote">"jenkins_event"</span>: <span class="code-quote">"job_run_queue_enter"</span>,
    <span class="code-quote">"blueocean_job_pipeline_name"</span>: <span class="code-quote">"blueocean-demo-project"</span>,
    <span class="code-quote">"jenkins_object_url"</span>: <span class="code-quote">"job/myfolder/job/blueocean-demo-project/job/feature%252Fnewhotness/"</span>,
    <span class="code-quote">"jenkins_channel"</span>: <span class="code-quote">"job"</span>
}
</pre>
</div></div>

<p>Right now this fails to update the UI because our UI logic refers to a pipeline by its fullName (which includes the folder) but our "blueocean_" props don't include the folder name. I also am not sure whether any of the comparisons on branch name are correct escaped right now. The "blueocean_job_rest_url" is also incorrect. The branch name needs to be escaped a second time. I also think that "blue" should be prepended on the front of the URL because that's how the HAL links are formatted.</p>

<p>What I think we need to do is ensure that the REST URL is formatted the same as the HAL links, and then do all of our comparisons against that value. </p>

</pre>
</details>

</ul>
</details>

<p>This may not be the right ticket for this comment but I want to be sure to capture it. Are we sure that the SSE events aren't actually coming through, or is it that the UI just isn't updating itself?</p>

<p>While debugging <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a> I noticed some issues with UI updates. SEE events were coming through ok after a run, but the UI was not updating. The SSE event we receive for a nested job looks like this:</p>


<pre>
{
    <span class="code-quote">"jenkins_object_type"</span>: <span class="code-quote">"org.jenkinsci.plugins.workflow.job.WorkflowJob"</span>,
    <span class="code-quote">"job_run_status"</span>: <span class="code-quote">"QUEUED"</span>,
    <span class="code-quote">"job_name"</span>: <span class="code-quote">"myfolder/blueocean-demo-project/feature%2Fnewhotness"</span>,
    <span class="code-quote">"jenkins_org"</span>: <span class="code-quote">"jenkins"</span>,
    <span class="code-quote">"job_run_queueId"</span>: <span class="code-quote">"23"</span>,
    <span class="code-quote">"job_ismultibranch"</span>: <span class="code-quote">"<span class="code-keyword">true</span>"</span>,
    <span class="code-quote">"jenkins_object_name"</span>: <span class="code-quote">"myfolder/blueocean-demo-project/feature%2Fnewhotness"</span>,
    <span class="code-quote">"blueocean_job_rest_url"</span>: <span class="code-quote">"/<span class="code-keyword">rest</span>/organizations/jenkins/pipelines/blueocean-demo-project/branches/feature%2Fnewhotness/"</span>,
    <span class="code-quote">"blueocean_job_branch_name"</span>: <span class="code-quote">"feature%2Fnewhotness"</span>,
    <span class="code-quote">"jenkins_event"</span>: <span class="code-quote">"job_run_queue_enter"</span>,
    <span class="code-quote">"blueocean_job_pipeline_name"</span>: <span class="code-quote">"blueocean-demo-project"</span>,
    <span class="code-quote">"jenkins_object_url"</span>: <span class="code-quote">"job/myfolder/job/blueocean-demo-project/job/feature%252Fnewhotness/"</span>,
    <span class="code-quote">"jenkins_channel"</span>: <span class="code-quote">"job"</span>
}
</pre>

<p>Right now this fails to update the UI because our UI logic refers to a pipeline by its fullName (which includes the folder) but our "blueocean_" props don't include the folder name. I also am not sure whether any of the comparisons on branch name are correct escaped right now. The "blueocean_job_rest_url" is also incorrect. The branch name needs to be escaped a second time. I also think that "blue" should be prepended on the front of the URL because that's how the HAL links are formatted.</p>

<p>What I think we need to do is ensure that the REST URL is formatted the same as the HAL links, and then do all of our comparisons against that value. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tfennelly.png" title="tfennelly's avatar" /> tfennelly:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=264126&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264126">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<blockquote><p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> Are we sure that the SSE events aren't actually coming through, or is it that the UI just isn't updating itself?</p></blockquote>

<p>Well it may not be ... in general, any error in the UI where things do not refresh properly gets logged in JIRA as "SSE is broken ....", so I just tend to read these JIRAs as "something is not refreshing in the UI" and it needs to be investigated to see what the real issue is.</p>

<p>Wrt the job url in the event message, that URL is coming from temp code in <tt>BlueMessageEnricher</tt> while we were waiting on <a href="https://issues.jenkins.io/browse/JENKINS-36286" title="Provide generic way to get Link for given Jenkins job name" class="issue-link" data-issue-key="JENKINS-36286"><del>JENKINS-36286</del></a> to be done. I added a comment wrt that in <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a>.</p>
</pre>
</details>

</ul>
</details>

<blockquote><p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=cliffmeyers" class="user-hover" rel="cliffmeyers">cliffmeyers</a> Are we sure that the SSE events aren't actually coming through, or is it that the UI just isn't updating itself?</p></blockquote>

<p>Well it may not be ... in general, any error in the UI where things do not refresh properly gets logged in JIRA as "SSE is broken ....", so I just tend to read these JIRAs as "something is not refreshing in the UI" and it needs to be investigated to see what the real issue is.</p>

<p>Wrt the job url in the event message, that URL is coming from temp code in <tt>BlueMessageEnricher</tt> while we were waiting on <a href="https://issues.jenkins.io/browse/JENKINS-36286" title="Provide generic way to get Link for given Jenkins job name" class="issue-link" data-issue-key="JENKINS-36286"><del>JENKINS-36286</del></a> to be done. I added a comment wrt that in <a href="https://issues.jenkins.io/browse/JENKINS-36615" title="Unable to run multibranch pipeline with '/' the name" class="issue-link" data-issue-key="JENKINS-36615"><del>JENKINS-36615</del></a>.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/cliffmeyers.png" title="cliffmeyers's avatar" /> cliffmeyers:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=264133&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-264133">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly" class="user-hover" rel="tfennelly">tfennelly</a> I believe you are back tomorrow? I can touch base w/ you then as to what I am working on. <a href="https://issues.jenkins.io/browse/JENKINS-33615" title="Admin rights delegation issue on W10 in powershell plugin" class="issue-link" data-issue-key="JENKINS-33615">JENKINS-33615</a> has some bugs and regressions dealing with nested jobs and branch names with special chars in them. That may effectively resolve this bug.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tfennelly" class="user-hover" rel="tfennelly">tfennelly</a> I believe you are back tomorrow? I can touch base w/ you then as to what I am working on. <a href="https://issues.jenkins.io/browse/JENKINS-33615" title="Admin rights delegation issue on W10 in powershell plugin" class="issue-link" data-issue-key="JENKINS-33615">JENKINS-33615</a> has some bugs and regressions dealing with nested jobs and branch names with special chars in them. That may effectively resolve this bug.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-36638?focusedId=265320&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-265320">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Wasn't reproduced locally and likely due to unupdated plugins on dogfood. </p>

<p>It is critical to reproduce things locally. If there is a problem with SSE and proxies it shoudl be clear and proven that is the case. </p>
</pre>
</details>

</ul>
</details>

<p>Wasn't reproduced locally and likely due to unupdated plugins on dogfood. </p>

<p>It is critical to reproduce things locally. If there is a problem with SSE and proxies it shoudl be clear and proven that is the case. </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `relates_to` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-36615%22">JENKINS-36615</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `epic` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-35737%22">JENKINS-35737</a>]</i>
